.github/workflows/build-node-packages.yml

```diff
@@ -0,0 +1,202 @@
+      shell: bash
+
+    - name: Fully cleanup the toolcache directory before testing
+      run: |
+        ./helpers/clean-toolcache.ps1 -ToolName "node"
+      shell: pwsh
+
+    - name: Download artifact
+      uses: actions/download-artifact@v2
+      with:
+        path: ${{ runner.temp }}
+
+    - name: Extract files
+      run: |
+        if ('${{ matrix.platform }}' -eq 'win32') {
+          $artifactName = "node-${{ env.VERSION }}-${{ matrix.platform }}-${{ env.ARCHITECTURE }}.7z"
+          7z.exe x "$artifactName" -y | Out-Null 
+        } else {
+          $artifactName = "node-${{ env.VERSION }}-${{ matrix.platform }}-${{ env.ARCHITECTURE }}.tar.gz"
+          tar -xzf $artifactName
+        }
+      working-directory: ${{ runner.temp }}/node-${{ env.VERSION }}-${{ matrix.platform }}
+      shell: pwsh
+
+    - name: Apply build artifact to the local machine
+      run: |
+        if ('${{ matrix.platform }}' -eq 'win32') { powershell ./setup.ps1 } else { sh ./setup.sh }
+      working-directory: ${{ runner.temp }}/node-${{ env.VERSION }}-${{ matrix.platform }}
+      shell: pwsh
+
+    - name: Setup node ${{ env.VERSION }}
+      uses: actions/setup-node@v2.1.1
+      with:
+        node-version: ${{ env.VERSION }}
+
+    - name: Wait for the logs
+      run: |
+        Write-Host "Fake step that do nothing"
+        Write-Host "We need it because log of previous step 'Setup Node' is not available here yet."
+        Write-Host "In testing step (Node.Tests.ps1) we analyze build log of 'Setup Node' task"
+        Write-Host "to determine if Node.js version was consumed from cache and was downloaded"
+      shell: pwsh
+
+    - name: Run tests
+      run: |
+        Install-Module Pester -Force -Scope CurrentUser -RequiredVersion 4.10.1
+        Import-Module Pester
+        $pesterParams = @{
+          Path="./Node.Tests.ps1";
+          Parameters=@{
+            Version="$env:VERSION";
+          }
+        }
+        Invoke-Pester -Script $pesterParams -EnableExit -OutputFile "test_results.xml" -OutputFormat NUnitXml
+      working-directory: ./tests
+      shell: pwsh
+
+  publish_release:
+    name: Publish release
+    if: github.event.inputs.PUBLISH_RELEASES == 'true'
+    needs: test_node
+    runs-on: ubuntu-latest
+    outputs:
+      upload_url: ${{ steps.create_release.outputs.upload_url }}
+    steps:
+    - name: Publish Release ${{ env.VERSION }}
+      id: create_release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        tag_name: ${{ env.VERSION }}-${{ github.run_id }}
+        release_name: ${{ env.VERSION }}
+        body: |
+          Upload Node.js ${{ env.VERSION }}
+
+  upload_assets:
+    name: Upload assets for ${{ matrix.platform }}
+    needs: publish_release
+    runs-on: ubuntu-latest
+    env:
+      UPLOAD_URL: ${{ needs.publish_release.outputs.upload_url }}
+    strategy:
+      matrix:
+        extension: ['tar.gz']
```

Comment:
This matrix looks very strange, can we describe all cases in include?

Reply:
removed matrix